### ISSUE TITLE
reflow and parentless or absolute-parented tasks

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#3737]
Should "reflow" mean ongoing spawning of child-tasks only, or does it also apply to parentless or absolute-parented tasks whose next instances get auto-spawned as they drop out of the runahead pool?   IMO its the latter, otherwise we can't trigger a reflow that is identical to the original flow in all of its tasks.  However, maybe ongoing auto-spawning needs to be optional. E.g.:

```
R1 = start
P1 = start[^] => foo => bar & baz => qux
```

If I retrigger `foo.6` with `--reflow` it should cause `bar.6` and `baz.6` then `qux.6` to run, but it will also result in `foo.7` being spawned and so on - is that to be expected or not? (Note we can get restricted child-only reflow by triggering `foo.6` without reflow, and then `cylc spawn foo.6` to spawn the children of `foo.6`).

**Comment by oliver-sanders** [comment]
If we had a workflow like this:

```ini
R1 = """
    # n = 1..3
    a<n> => b<n> => c<n> & d<n>
"""
```

And we did `cylc trigger --reflow <flow> b_2.1` then I think the reflow should only effect the downstream tasks like so:

![*-1](https://user-images.githubusercontent.com/16705946/109684721-75592300-7b78-11eb-898f-8b8084e01d15.png)

There is no relation between `b_2` and `b_3` so we wouldn't expect these parallel graphs to interact.

I don't think swapping parameters for cycles should make any difference in this case difference:

```ini
P1 = """
    a => b => c & d
"""
```

![*-0](https://user-images.githubusercontent.com/16705946/109685276-ff08f080-7b78-11eb-9b71-b0f0117abdbb.png)

**Comment by hjoliver** [comment]
Agreed, but my question really only relates to triggering (in your example) `a.2` (say), which has no parents. Then, with `--reflow` should `a.3` (and so on) get spawned too?

We might need to both. E.g. default to strictly graph-children; but include auto-spawned tasks as well with some new option flag.

**Comment by hjoliver** [comment]
Clear documentation required!!

**Comment by oliver-sanders** [comment]
@hjoliver can this become a cylc-doc issue now?

**Comment by hjoliver** [comment]
It's should be easy to add an option to prevent flow-on to subsequent cycles.  I'll assume we want that (I believe this issue was motivated by you and/or Dave thinking it shouldn't flow on) and remove the question. Lemme know quickly if you've changed your mind on that.

**Comment by oliver-sanders** [comment]
I think with the retiring of the term "reflow" (#4686) the current behaviour is easier to understand (although we do need to document it!).

More restrictive triggering can follow as required, would rather not muddy the waters with complexity at this point.

Happy to keep current approach and transfer this issue to cylc-doc.
